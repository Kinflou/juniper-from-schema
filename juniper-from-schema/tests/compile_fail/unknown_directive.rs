#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../compile_pass/setup.rs");

juniper_from_schema::graphql_schema! {
    type Query {
        string: String! @someDirectiveThatIsntNotSupported
    }

    schema { query: Query }
}

pub struct Query;

impl QueryFields for Query {
    fn field_string(&self, executor: &Executor<Context>) -> FieldResult<&String> {
        unimplemented!()
    }
}
