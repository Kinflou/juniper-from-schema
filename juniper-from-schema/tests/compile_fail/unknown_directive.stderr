error: proc macro panicked
  --> $DIR/unknown_directive.rs:4:1
   |
4  | / juniper_from_schema::graphql_schema! {
5  | |     type Query {
6  | |         string: String! @someDirectiveThatIsntNotSupported
7  | |     }
8  | |
9  | |     schema { query: Query }
10 | | }
   | |_^
   |
   = help: message:

           error: Unknown directive
            --> schema:1:32
             |
           1 |    type Query { string : String ! @ someDirectiveThatIsntNotSupported } schema
             |                                   ^

           aborting due to previous error

error[E0405]: cannot find trait `QueryFields` in this scope
  --> $DIR/unknown_directive.rs:14:6
   |
14 | impl QueryFields for Query {
   |      ^^^^^^^^^^^ not found in this scope
